javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/matcher/AgentElementMatchers.java

```diff
@@ -25,8 +25,7 @@
 
   public static ElementMatcher.Junction<TypeDescription> implementsInterface(
       ElementMatcher<TypeDescription> matcher) {
-    return not(isInterface())
-        .and(new SafeHasSuperTypeMatcher(new SafeErasureMatcher<>(matcher), true));
+    return new SafeHasSuperTypeMatcher(new SafeErasureMatcher<>(matcher), true);
```

Comment:
this change was not needed for JAX-RS default methods, but I looked through callers and not sure why we wouldn't also want to instrument interface default methods, so removed `not(isInterface())` here too